### PR TITLE
Lol Message Blog

### PR DESCRIPTION
**Pull request by n4ze3m**

### CODE REVIEW

File: lol_message/blog.md [hunk 1]
```diff
@@ -0,0 +1,571 @@
+# Lol Message
```
Comment (shreythecray) [inline lol_message/blog.md:1]: Great job, Nazeem!

File: lol_message/blog.md [hunk 2]
```diff
@@ -0,0 +1,571 @@
...
+
+
+## Background
+
+During the [Courier Hack](https://courier-hacks.devpost.com/), I wanted to build a project that was fun and easy to use. So I decided to build an open source `NGL.link` alternative with email notification and a dashboard to view all the received messages.
```
Comment (shreythecray) [inline lol_message/blog.md]: Here are some suggestions to add to your introduction:
- what is NGL.link and why is that a good example to follow?
- what is the value of making this project open source?
- why did you choose email notifications over other channels?
- what are the capabilities of this application?

I would also recommend embedding or adding a link to your demo video here, and linking your DevPost page in the Quick Links section below.

File: lol_message/blog.md [hunk 3]
```diff
@@ -0,0 +1,571 @@
...
+
+I decided to use the following tools and technologies for this project:
+
+- Notification: [Courier](https://courier.com/)
+- Web Framework: [Next.js](https://nextjs.org/)
+- Backend Framework: [tRPC](https://trpc.io/)
+- Authentication & Database: [Supabase](https://supabase.io/)
+- UI: [Mantine](https://mantine.dev/)
+
+## Instructions
```
Comment (shreythecray) [inline lol_message/blog.md:28]: - Are there any pre-reqs to building this project?
- Are there are skills that the user should have to successfully complete this or can people with little-no experience complete it?

File: lol_message/blog.md [hunk 4]
```diff
@@ -0,0 +1,571 @@
...
+
+We will be building core features of the project in this tutorial. You can find the source code of the full project [here](https://github.com/n4ze3m/lol), and you can find the live demo [here](https://lol-message.vercel.app/).
```
Comment (shreythecray) [inline lol_message/blog.md]: Love this!

File: lol_message/blog.md [hunk 5]
```diff
@@ -0,0 +1,571 @@
...
+
+So let's get started!
+
+Below diagram shows the architecture which we will be building in this article.
+
+![Architecture Diagram](https://i.imgur.com/WtTIASB.png)
```
Comment (shreythecray) [inline lol_message/blog.md:36]: This is fantastic! You should explain this diagram in writing for increased accessibility.

File: lol_message/blog.md [hunk 6]
```diff
@@ -0,0 +1,571 @@
...
+Next, we need to create a new prisma schema file. You can use the following command to create a new prisma schema file:
+
+```bash
+yarn prisma init
+```
+
+After creating the prisma schema file, we need to add the following code to the `schema.prisma` file:
+
+```prisma
+generator client {
```
Comment (shreythecray) [inline lol_message/blog.md:148]: I'd recommend breaking this down into four code snippets and explanations for each of them:

- generator client
- datasource db
- model user
- model message

File: lol_message/blog.md [hunk 7]
```diff
@@ -0,0 +1,571 @@
...
+  });
+
+if (process.env.NODE_ENV !== "production") global.prisma = database;
+```
+
+The above code is used to connect to the database using `prisma` client.
+
+Next open your trpc router file and add the following code:
+
+```ts
```
Comment (shreythecray) [inline lol_message/blog.md:310]: Let's break this down too. It can be a lot for someone who has never done this before.

**Review by shreythecray — CHANGES_REQUESTED** [review]
This is amazing, Nazeem! I've added some comments. Once you have resolved them all, I'll take it from there and make some edits before merging.

File: lol_message/blog.md (hunk 1 above)
Comment (n4ze3m) [reply lol_message/blog.md:1]: Thanks

**Pull request merged**